src/main/java/org/folio/rest/impl/InvoiceStorageImpl.java

```diff
@@ -158,6 +174,140 @@ public void putInvoiceStorageInvoicesById(String id, String lang, Invoice entity
+      try {
+        Criterion criterion = getCriterionByFieldNameAndValue(INVOICE_ID_FIELD_NAME, id);
+        pgClient.get(DOCUMENT_TABLE, Document.class, criterion, false, reply -> {
+          try {
+            if (reply.succeeded()) {
+
+              List<Document> results = reply.result().getResults();
+
+              DocumentCollection collection = new DocumentCollection();
+              collection.setDocuments(results);
+
+              Integer totalRecords = reply.result().getResults().size();
+              collection.setTotalRecords(totalRecords);
+
+              asyncResultHandler.handle(Future.succeededFuture(GetInvoiceStorageInvoicesDocumentsByIdResponse.respond200WithApplicationJson(collection)));
+            } else {
+              log.error(reply.cause().getMessage(), reply.cause());
+              asyncResultHandler
+                .handle(Future.succeededFuture(GetInvoiceStorageInvoicesDocumentsByIdResponse.respond500WithTextPlain(reply.cause().getMessage())));
+            }
+          } catch (Exception e) {
+            log.error(e.getMessage(), e);
+            asyncResultHandler.handle(Future.succeededFuture(GetInvoiceStorageInvoicesDocumentsByIdResponse.respond500WithTextPlain(Response.Status.INTERNAL_SERVER_ERROR.getReasonPhrase())));
+          }
+        });
+      } catch (Exception e) {
+        log.error(e.getMessage(), e);
+        asyncResultHandler.handle(Future.succeededFuture(GetInvoiceStorageInvoicesDocumentsByIdResponse.respond500WithTextPlain(Response.Status.INTERNAL_SERVER_ERROR.getReasonPhrase())));
+      }
+    });
+  }
+
+  @Validate
+  @Override
+  public void postInvoiceStorageInvoicesDocumentsById(String id, String lang, InvoiceDocument entity, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    vertxContext.runOnContext((Void v) ->{
+      if (!StringUtils.equals(entity.getDocumentMetadata().getInvoiceId(), id)) {
+        asyncResultHandler.handle(Future.succeededFuture(PostInvoiceStorageInvoicesDocumentsByIdResponse.respond400WithTextPlain(INVOICE_ID_MISMATCH_ERROR_MESSAGE)));
+        return;
+      }
+
+      pgClient.getClient().getConnection(sqlConnection -> {
+        try {
+
+          String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT));
+          String fullTableName = PostgresClient.convertToPsqlStandard(tenantId) + "." + DOCUMENT_TABLE;
+
+          entity.getDocumentMetadata().setMetadata(entity.getMetadata());
+
+          boolean base64Exists = entity.getContents() != null && StringUtils.isNotEmpty(entity.getContents().getData());
+          String sql = "INSERT INTO " + fullTableName + " (id, " + (base64Exists ? "document_data," : "") + " jsonb) VALUES (?," + (base64Exists ? "?," : "") + " ?::JSON) RETURNING id";
+          JsonArray jsonArray = prepareDocumentQueryParams(entity);
+
+          sqlConnection.result().queryWithParams(sql, jsonArray, query -> {
```

Comment:
Please consider to use `pgClient.execute(sqlConnection, sql, jsonArray, <replyHandler>)`

Reply:
Implementation was updated to use `pgClient.execute`